R/format_PEW_temp.R

```diff
@@ -0,0 +1,541 @@
+                                               "text", "numeric", "numeric", "numeric",
+                                               "numeric", "numeric", "numeric",
+                                               "numeric", "numeric", "numeric")) %>%
+    janitor::clean_names(case = "upper_camel") %>%
+    janitor::remove_empty(which = "rows") %>%
+    #### Solve NA values (some are explicitly stated as character "NA")
+    dplyr::mutate(across(where(is.character), .fns = ~replace(., . ==  "NA" , NA))) %>%
+    #### Convert to corresponding format and rename
+    dplyr::mutate(BreedingSeason = as.integer(Year),
+                  CaptureDate = as.Date(Date),
+                  Tarsus = as.numeric(Tarsus),
+                  Mass = as.numeric(Mass),
+                  TarsusPartner = as.numeric(TarsusPartner),
+                  MassPartner = as.numeric(MassPartner),
+                  ClutchSize = as.integer(ClutchSize),
+                  DateEgg1 = as.Date(DateEgg1),
+                  HatchDateD0 = as.Date(HatchDateD0),
+                  NumberOfRingedChicks = as.integer(NumberOfRingedChicks),
+                  DateRingingChicks = as.Date(DateRingingChicks),
+                  NoOfChicksD3 = as.integer(NoOfChicksD3),
+                  BroodMassD3 = as.numeric(BroodMassD3),
+                  ObservationTimeH = as.numeric(ObservationTimeH),
+                  Species = "CYACAE",
+                  PopID = "PEW",
+                  NestboxID = tolower(Nest),
+                  BroodID = ifelse(Method %in% c("Catch adults nestbox", "ChickRinging"),
+                                   paste(Year, NestboxID, "PEW", sep = "_"), NA)) %>%
+    #### Rename variables to standardized format
+    dplyr::rename(IndvID = Id,
+                  Sex =  Seks,
+                  ObserverID = Measurer) %>%
+    #### Remove columns which we do not store in the standardized format
+    dplyr::select(-FeatherCollection ,
+                  -BreathRateTime50Breaths,
+                  -FeathersPartner,
+                  -BreathRatePartnerTime50Breaths,
+                  -BloodSample,
+                  -TimeBloodSample,
+                  -BloodSampleDuration,
+                  -VisitRateVisitsH,
+                  -VrPartner,
+                  -VisitsAlternated,
+                  -VisitsAlternatedPartner,
+                  -VisitsSync10,
+                  -ChickAgeOfBehavObserv,
+                  -MateStrategy,
+                  -Nest) %>%
+    #### Reorder columns
+    dplyr::select(BreedingSeason,
+                  Species,
+                  PopID,
+                  IndvID,
+                  Sex,
+                  Age,
+                  everything()) %>%
+    #### Remove observations without ID
+    dplyr::filter(!is.na(IndvID)) %>%
+    #### Details
+    dplyr::mutate(PartnerId = ifelse(PartnerId == "MetalRing_FemK89_2019",
+                                     "MetalRing_F2019_K89", PartnerId))
```

Comment:
This might get tricky, as there are several forms of coding for unringed individuals. I first though to write some generalizable code to unify them, but after I only found the one specific individual written in two different forms, I transformed that one in particular.

The most common ways used are:
Unknow_SexNest_Year (Unknown_Male75_2017)
Unknow_SexYearNest (seems to be a version used only in the year 2019: Unknown_M2019NestK97)
u/Unringed_SexNest_Year (unringed_Nest103_2017)
UniParental (or semiparental or Semi-parental or SemiUniParental)_SexNest_Year (UniParental_MaleK99_2016)
MetalRing or NoMetalRing_SexYearNest (MetalRing_M2019Nest103, NoMetalRing_M2019NestK110)

If you think it would be good to unify them, do you have in mind any specfic format? Maybe we could remove the first word (unringed, unknow,...) and keep only the sex, nest and year in unified format, for example Sex_NestID_Year?